### ISSUE TITLE
Automated documentation previews in pull requests

### ISSUE THREAD

**Issue by weiji14** [GenericMappingTools/pygmt#342]
**Description of the desired feature**

Make it easier to preview changes to the documentation **online** in every pull request, rather than having to clone the branch and do a **local** build. I.e. Continuous Documentation.

Will be especially helpful for new contributors and even reviewers wanting a quick look at what's going up on the website. This was hinted at during #314.

![Documentation preview](https://assets.zeit.co/image/upload/q_auto/v1567540580/front/blog/deploy-summary/blogpost-image-03.png)

I'm gonna try to set up an automated Continuous Deployment site, just a heads up!

**Are you willing to help implement and maintain this feature?** Yes

**Comment by seisman** [comment]
@weiji14 I received your request to install a github app to the pygmt repository, but I can't find any description about the app. Is it related to this issue? Could you post the homepage of this app?

**Comment by weiji14** [comment]
Here it is https://zeit.co/.

**Comment by weiji14** [comment]
I'll try to get it to work on my own fork first. Need to find a way to make it do `cd doc && make all`

**Comment by seisman** [comment]
I’ve approved and installed the app.

**Comment by weiji14** [comment]
Cool, this might take a bit of time (and experimentation) to set up, but hope it'll be worth it!

**Comment by weiji14** [comment]
Hmm, I might have messed up the github app integration. Sorry @seisman, could you uninstall it first, and I'll try to request it to be reinstalled again?

**Comment by seisman** [comment]
@weiji14 I've uninstalled it.

**Comment by seisman** [comment]
@weiji14 I've approved the app and send you an email with a Now link, which may be necessary for installation.

**Comment by weiji14** [comment]
Ah cool, I think I've got it up now. I've also emailed you an invite link to join the 'gmt' organization on [Now](https://zeit.co). There's shouldn't be much to configure though, getting the Github App Integration set up should be all we need.

If you can review #343, I'll get that merged and close this issue in #344.

**Comment by leouieda** [comment]
Thanks, Wei! I experimented with this in the past but it was hard to do with the CIs because forks don’t have access to encrypted variables needed for deploying. Does this app offer a way around that?

**Comment by weiji14** [comment]
The [Zeit Now for Github App](https://zeit.co/github) integration does a deploy every time a `git push` is made to the repo (be it in a Pull Request or master), and I think it has certain permissions to allow it to do that. For example at #344, there's a url for every commit, and also one for the PR/fork that keeps itself updated every commit, and is referenced to the fork's name (e.g. https://pygmt-git-fork-weiji14-continuous-docs.gmt.now.sh).

I did experiment with making deployments via Github Actions CI (which I've attempted in my personal fork at https://github.com/weiji14/pygmt/pull/45) but that required setting a 'secrets' token variable. We *can* use a token as well to deploy to Zeit Now, but it's not strictly required with the app.
